sky/compositor/picture_layer.cc

```diff
@@ -14,22 +14,17 @@ PictureLayer::PictureLayer() {
 PictureLayer::~PictureLayer() {
 }
 
-SkMatrix PictureLayer::model_view_matrix(const SkMatrix& model_matrix) const {
-  SkMatrix modelView = model_matrix;
-  modelView.postTranslate(offset_.x(), offset_.y());
-  return modelView;
-}
-
 void PictureLayer::Paint(PaintContext::ScopedFrame& frame) {
   DCHECK(picture_);
 
   const SkRect& bounds = paint_bounds();
-  SkISize size = SkISize::Make(bounds.width(), bounds.height());
-
-  RefPtr<SkImage> image =
-      frame.paint_context().rasterizer().GetCachedImageIfPresent(
-          frame.paint_context(), frame.gr_context(), picture_.get(), size);
+  const SkISize size = SkISize::Make(bounds.width(), bounds.height());
```

Comment:
Here you've truncated from floats to ints.  We need to wait until we've multiplied in the matrix before doing that to avoid losing precision.  Consider a size of 12.5 and a scale of 2x.  We want to end up with 25 pixels instead of 24 or 26.